util/drakeUtil.h

```diff
@@ -84,19 +93,86 @@ mxArray* stdVectorToMatlab(const std::vector<Scalar>& vec) {
   return pm;
 }
 
-DLLEXPORT const std::vector<double> matlabToStdVector(const mxArray* in);
+DLLEXPORT void sizecheck(const mxArray* mat, int M, int N);
+
+template <size_t Rows, size_t Cols>
```

Comment:
agreed.  but you're just passing in double* everywhere I've seen this called, so I don't think you're getting any extra safety from the notation here.  and we're adding template complexity, increasing compile time, and memory footprint.